pkg/operator/etcdendpointscontroller/etcdendpointscontroller.go

```diff
@@ -83,9 +83,30 @@ func (c *EtcdEndpointsController) sync(ctx context.Context, syncCtx factory.Sync
 	return nil
 }
 
-func (c *EtcdEndpointsController) syncConfigMap(ctx context.Context, recorder events.Recorder) error {
+func (c *EtcdEndpointsController) syncConfigMap(recorder events.Recorder) error {
+	bootstrapComplete, err := isBootstrapComplete(c.configmapLister, c.operatorClient)
+	if err != nil {
+		return fmt.Errorf("couldn't determine bootstrap status: %w", err)
+	}
+
 	required := configMapAsset()
 
+	// If the bootstrap IP is present on the existing configmap, either copy it
+	// forward or remove it if possible so clients can forget about it.
+	if existing, err := c.configmapLister.ConfigMaps(operatorclient.TargetNamespace).Get("etcd-endpoints"); err == nil {
+		if existingIP, hasExistingIP := existing.Annotations[etcdcli.BootstrapIPAnnotationKey]; hasExistingIP {
+			if bootstrapComplete {
+				// remove the annotation
+				recorder.Eventf("EtcdEndpointsRemovedBootstrapIPAnnotation", "removed the bootstrap IP %q from configmap %s/%s", existingIP, existing.Namespace, existing.Name)
```

Comment:
Just a general comment on event workflow it seems wrong that we event something as valid before it succeeds. For example if apply failed below event review would appear to show a success.

Reply:
Good catch, agree

Reply:
I went ahead and removed the event entirely, I'm not sure it's really useful anymore